### ISSUE TITLE
[feature request] Windows version and media tags

### ISSUE THREAD

**Issue by dardoor** [kimono-koans/dano#3]
Hi!

Is there a chance for a Windows version?

And it would be nice if the checksum was saved into a standard media tag (same as 'title', 'author', etc). If it worked with MKV it'd already be pretty good, no need to support every format. That would make it easier to see the data with other tools and be OS agnostic.

**Comment by kimono-koans** [comment]
Thanks for filing an issue.  I understand you as having two requests:

First, a Windows version.  A Windows version is not a priority to me, because I don't use Windows (Mac/Linux only household I'm afraid).  However, I'd be very pleased to include support a Windows version, if someone else wanted to submit a PR.  

For anyone interested, I actually don't think this would be that hard to do.  AFAIK is nothing about `dano` that wouldn't work on Windows (`ffmpeg` works on Windows right?).  Just try to build (it may just work?) and fix the errors (a few well placed: if cfg!(target_os = "windows"), etc.).  Someone with minimal Rust experience could do it in a few hours.  I could probably build this quickly, but I don't want to include support for anything no one is testing.

Second, writing to a media tag instead of file/xattr.  I'd be pleased to add support for this, if someone could show me how to write tags "in place" with `ffmpeg` (without writing an output file), or otherwise describe another file format agnostic way of doing what you describe (best case: a Rust crate with in place tag writing support).  

Put simply -- I want to just be able to blast over files without thinking about whether this is a mp3 or FLAC file.  This program is already more complex than I thought it would be.  I won't be adding any more additional complexity than necessary.  There should be "one way" to write/read a tag, just like an xattr is one way to write/read a tag. 

So -- Let me know if you have any really good ideas on tags!  I'll keep this open for you and others to add your thoughts.

Thanks again!

**Comment by dardoor** [comment]
FWIW, there's a [foobar2000 component](https://foobar.hyv.fi/?view=foo_audiomd5) which can do this, but it only works for audio.

**Comment by kimono-koans** [comment]
> As long as xattr is a dependency, the project cannot compile on Windows—this applies to both the current version 1.3 and the latest 1.6. Therefore, unless the crate is replaced or modified, it seems Windows compilation is not possible at this time. I just wanted to share this observation in case it helps with planning Windows support.

Perhaps that makes this a good future project for you (when you learn more Rust!), or for another Windows devotee.   

AFAICT NTFS *does* support extended attributes, they simply aren't used for much.  The easiest method to add support to dano might be to consider adding NTFS support to xattr?  I know the maintainer of xattr a little, and I wouldn't approach him/her with this until you were ready to do the work.  xattr is a low level library, so plan to learn Rust for a few years, but when you are ready to contribute to a library, that might be an interesting feature, for you or someone else, to contribute to xattr.

**Comment by God-damnit-all** [comment]
I was hoping to use this on Windows for the main feature, the hashing of streams rather than files. Any chance of making it so the media tag feature is simply omitted on native Windows builds?

**Comment by kimono-koans** [comment]
> I was hoping to use this on Windows for the main feature, the hashing of streams rather than files. Any chance of making it so the media tag feature is simply omitted on native Windows builds?

@God-damnit-all I would be very pleased to have a Windows version of `dano`, even a limited version.  

The most pressing issue is I don't use Windows!  So -- for what you are asking -- someone would need to develop support for Windows, test it on Windows, and promise me to support their Windows features into the future.

You or someone else are more than welcome to have a go, but I won't be doing this myself because -- as I said -- I don't use Windows.
